packages/cubejs-vue/src/QueryRenderer.js

```diff
@@ -41,40 +44,50 @@ export default Vue.component('QueryRenderer', {
     }
   },
   render(createElement) {
-    const { resultSet, error, loading, sqlQuery } = this;
+    const { $scopedSlots, resultSet, error, loading, sqlQuery } = this;
+    const empty = createElement('div', { class: { 'no-content': true }});
+    let slot = this.$slots.empty ? this.$slots.empty : empty;
+    let controls = createElement('div', { class: { 'no-information': true }});
+
+    if ($scopedSlots.builder && this.builderProps.measures) {
+      controls = $scopedSlots.builder({ ...this.builderProps });
+    }
 
-    if (!loading && resultSet) {
+    if (!loading && resultSet && !error) {
       const slotProps = {
-        resultSet: this.queries ? (resultSet || {}) : resultSet,
-        error,
-        isLoading: loading,
+        resultSet,
         sqlQuery,
+        query: this.builderProps.query,
+        loading,
+        error,
         ...this.builderProps,
       };
 
-      return createElement(
-        'div',
-        this.$scopedSlots.default(slotProps),
-      );
-    } else {
-
-      return createElement(
-        'div',
-        this.$scopedSlots.default({
-          loading,
-          error,
-          sqlQuery,
-          ...this.builderProps,
-        }),
-      );
+      slot = $scopedSlots.default ? $scopedSlots.default(slotProps) : slot;
+    } else if (error) {
+      slot = $scopedSlots.error ? $scopedSlots.error({ error, sqlQuery }) : slot;
```

Comment:
Do you think we should fallback here to default if only one slot is passed for backward compatibility?

Reply:
You're right i missed that one